src/systems/userinput/userinput.js

```diff
@@ -165,21 +247,46 @@ AFRAME.registerSystem("userinput", {
   },
 
   tick() {
-    updateActionSetsBasedOnSuperhands();
+    const registeredMappingsChanged = this.registeredMappingsChanged;
+    if (registeredMappingsChanged) {
+      this.registeredMappingsChanged = false;
+      this.prevSortedBindings = this.sortedBindings;
+      this.sortedBindings = dependencySort(this.registeredMappings);
+      if (!this.prevSortedBindings) {
+        this.prevSortedBindings = this.sortedBindings;
+      }
+      this.masks = computeMasks(this.sortedBindings);
+    }
 
+    resolveActionSets();
     for (const { set, value } of this.pendingSetChanges) {
       this.activeSets[value ? "add" : "delete"](set);
     }
+    const activeSetsChanged = this.pendingSetChanges.length; // TODO: correct this
```

Comment:
Need to check if the active sets from last frame != the active sets from this frame